R/transform_runner.R

```diff
@@ -50,24 +50,9 @@ verify_data <- function(con, transformed_data) {
   if (length(transformed_data) == 0) {
     stop("Data transform failed, returned empty list.")
   }
-  for (table_name in names(transformed_data)) {
-    if (!DBI::dbExistsTable(con, table_name)) {
-      stop(sprintf(
-        "Table '%s' returned by transform but is missing from db schema.",
-        table_name
-      ))
-    }
-    col_names <- DBI::dbListFields(con, table_name)
-    for(col_name in colnames(transformed_data[[table_name]])) {
-      if(!(col_name %in% col_names)) {
-        stop(sprintf(
-          "Column '%s' in table '%s' returned by transform but is missing from db schema.",
-          col_name,
-          table_name
-        ))
-      }
-    }
-  }
-  ## TODO: Also check each table doesn't violate any constraints
+  lapply(names(transformed_data), function(x) {
```

Comment:
As a style thing, I'm not a big fan of `lapply` that ignores return values and would probably tend to write this as:

```
for (x in names(transformed_data)) {
 verify_table(con, x, transformed_data[[x]],
             context_info = "Transformed data")
}
```

as that makes it clearer (IMO) that we're calling `verify_table` purely for its side effects

Reply:
That makes sense, I've updated that.